### PR TITLE
Homework3: Eldar Dniaprouski

### PR DESCRIPTION
**Pull request by eldar-dniaprouski**

### CODE REVIEW

File: src/Homework_03/01_FizzBuzz.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+"""
+Напишите программу, которая печатает цифры от 1 до 100,
+но вместо чисел, кратных 3 пишет Fizz,
+вместо чисел кратный 5 пишет Buzz,
+а вместо чисел одновременно кратных и 3 и 5 - FizzBuzz
+"""
+
+for num in range(1, 101):
+    if not num % 3 and not num % 5:
```
Comment (belovmd) [inline src/Homework_03/01_FizzBuzz.py]: 15?

File: src/Homework_03/02_list_practice.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+['1a', '2a', '3a', '4a'].
+Одной строкой удалите элемент  '2a' из прошлого списка и напечатайте его.
+Скопируйте список и добавьте в него элемент '2a' так
+чтобы в исходном списке этого элемента не было.
+"""
+
+my_first_list = [a + b for a in 'ab' for b in 'bcd']
+print(my_first_list)
+print(my_first_list[::2])
+my_second_list = [str(num) + symb for num in range(1, 5) for symb in 'a']
```
Comment (belovmd) [inline src/Homework_03/02_list_practice.py]: `+'a'` ?

File: src/Homework_03/03_tuple_practice.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+"""
+Создайте список ['a', 'b', 'c'] и сделайте из него кортеж.
+Создайте кортеж ('a', 'b', 'c'), И сделайте из него список
+Сделайте следующие присвоения одной строкой a = 'a', b=2, c=’python’.
+Создайте кортеж из одного элемента, чтобы при итерировании по этому элементы
+последовательно выводились значения 1, 2, 3.
+Убедитесь что len() исходного кортежа возвращает 1.
+"""
+
+my_first_list = [a for a in 'abc']
```
Comment (belovmd) [inline src/Homework_03/03_tuple_practice.py]: a - bad name

File: src/Homework_03/03_tuple_practice.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+print(my_first_list)
+my_first_tuple = tuple(my_first_list)
+print(my_first_tuple)
+my_second_tuple = tuple(b for b in 'abc')
```
Comment (belovmd) [inline src/Homework_03/03_tuple_practice.py]: b - bad naming

File: src/Homework_03/04_pary_elementov.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+Важно: 1 1 1 - это 3 пары, 1 1 1 1 - это 6 пар
+"""
+
+num_string = '1 1 1 2 2 2 1'
+num_list = num_string.split()
+num_pair = 0
+check_list = num_list[:]
+for elem in num_list:
+    check_list.remove(elem)
+    for item in check_list:
```
Comment (belovmd) [inline src/Homework_03/04_pary_elementov.py]: O(n**2) and it could be solved as O(n)

File: src/Homework_03/07_chocolate.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+elif cuts >= pieces and \
+        (pieces > choc_sum - length or pieces > choc_sum - width):
+    print('Можно отломить {pieces} долек за {cuts} разломов'.
+          format(pieces=pieces, cuts=cuts))
+elif cuts >= pieces + 1:
+    print('Можно отломить {pieces} долек за {cuts} разломов'.
+          format(pieces=pieces, cuts=cuts))
+else:
+    print('Нельзя отломить {pieces} долек за {cuts} разломов'.
+          format(pieces=pieces, cuts=cuts))
```
Comment (belovmd) [inline src/Homework_03/07_chocolate.py:59]: missed third part:)

File: src/Homework_03/dop_summa_sosedei.py [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+Вывoд должен содержать одну строку с чиcлами новoго списка,
+разделёнными пробeлом.
+"""
+
+num_str = input('Введите список чисел через пробел\n')
+numbers = num_str.split()
+
+#  Вариант 1 Не совсем правильный
+my_str = ''
+for i in range(len(numbers)):
```
Comment (belovmd) [inline src/Homework_03/dop_summa_sosedei.py]: range(len(num))

File: src/Homework_03/dop_summa_sosedei.py [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+        cur_num = int(numbers[i + 1]) + int(numbers[i - 1])
+        my_str += str(cur_num) + ' '
+print(my_str)
+
+#  Вариант 2 Более правильный, но более длинный
+my_str2 = ''
+for elem in numbers:
+    if len(numbers) == 1:
+        my_str2 = elem
+    elif not numbers.index(elem):
```
Comment (belovmd) [inline src/Homework_03/dop_summa_sosedei.py:21]: o(n**2) it couldn't be more correct ^)

**Review by belovmd — COMMENTED** [review]
Please address these issues

File: src/Homework_03/01_FizzBuzz.py (hunk 1 above)
Comment (eldar-dniaprouski) [reply src/Homework_03/01_FizzBuzz.py]: Didn't think about that) Done

File: src/Homework_03/02_list_practice.py (hunk 2 above)
Comment (eldar-dniaprouski) [reply src/Homework_03/02_list_practice.py]: Got it

File: src/Homework_03/03_tuple_practice.py (hunk 3 above)
Comment (eldar-dniaprouski) [reply src/Homework_03/03_tuple_practice.py]: Fixed

File: src/Homework_03/03_tuple_practice.py (hunk 4 above)
Comment (eldar-dniaprouski) [reply src/Homework_03/03_tuple_practice.py]: Fixed

File: src/Homework_03/04_pary_elementov.py (hunk 5 above)
Comment (eldar-dniaprouski) [reply src/Homework_03/04_pary_elementov.py]: Solved as O(n)

File: src/Homework_03/07_chocolate.py (hunk 6 above)
Comment (eldar-dniaprouski) [reply src/Homework_03/07_chocolate.py:59]: Isn't this the third part?

File: src/Homework_03/dop_summa_sosedei.py (hunk 7 above)
Comment (eldar-dniaprouski) [reply src/Homework_03/dop_summa_sosedei.py]: Removed

File: src/Homework_03/dop_summa_sosedei.py (hunk 8 above)
Comment (eldar-dniaprouski) [reply src/Homework_03/dop_summa_sosedei.py:21]: So, it's the right solution?)

**Review by belovmd — COMMENTED** [review]
9

**Pull request closed without merging**